File: engine/engine-model/src/main/scala/tools/packet/pin/PinCodePacket.scala
package tools.packet.pin

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class PinCodePacket(private var _operation: PinOperation) extends PacketInput {
  def operation: PinOperation = _operation

  override def opcode(): SendOpcode = SendOpcode.CHECK_PIN_CODE
}

File: engine/engine-model/src/main/java/tools/packet/pin/PinCodePacket.java
package tools.packet.pin;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record PinCodePacket(PinOperation operation) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.CHECK_PIN_CODE;
   }
}